### ISSUE TITLE
feature request: tracking on domoticz log dzvent script name

### ISSUE THREAD

**Issue by megamarco833** [domoticz/domoticz#6233]
hello,
the feature request is to understand inside domoticz logs what is the dzvents script that is triggering an event
example from domoticz log:

`2024-12-26 11:10:21.534 Status: EventSystem: Script event triggered: /home/pi/domoticz/dzVents/runtime/dzVents.lua`

but if i have multiple dzvents script i don't know who is the dzvent script that trigger that event
same if you look at devices log:
![immagine](https://github.com/user-attachments/assets/6136e104-1139-406a-b079-94700bca5893)

i know that dzvent scripts are an internal process of domoticz, so with actual managment it's impossible to track the dzvent script name inside domoticz log, but in the future do you think that could be managed in some way?

**Comment by daserra23** [comment]
This would be excellent and help a lot with debugging.

**Comment by megamarco833** [comment]
> This would be excellent and help a lot with debugging.

Yes, it is, but @gizmocuz i don't think it will be a simple improvement due to the fact that dzvents is internal process of domoticz.. Or it could be feasible?

**Comment by gizmocuz** [comment]
It won't be a easy fix. I think we have to modify the dzVents Lua runtime file and some call functions.
But I agree, it would be very nice to have this instead of a global dzVents name

**Comment by megamarco833** [comment]
> It won't be a easy fix. I think we have to modify the dzVents Lua runtime file and some call functions. But I agree, it would be very nice to have this instead of a global dzVents name

if someone could implement this feature it would be very nice.
get a look at this example:
`2025-02-03 04:11:11.756 Error: EventSystem: Warning!, lua script /home/pi/domoticz/dzVents/runtime/dzVents.lua has been running for more than 10 seconds`
in that case i can't understand which is the dzvent script that fails

**Comment by megamarco833** [comment]
> It won't be a easy fix. I think we have to modify the dzVents Lua runtime file and some call functions. But I agree, it would be very nice to have this instead of a global dzVents name

hello, do you think that you could spend some time on it? :)

**Comment by megamarco833** [comment]
> It won't be a easy fix. I think we have to modify the dzVents Lua runtime file and some call functions. But I agree, it would be very nice to have this instead of a global dzVents name

hello @gizmocuz and hapy new year :)
do you think that will be possible for you to get a look / modify the dzVents Lua runtime file and some call functions to see in the log who is the script that is triggering the event? (example in my first post)

**Comment by Zuko789** [comment]
Hi! Is this feature enhancement available to work on or has it already been resolved?

**Comment by gizmocuz** [comment]
Are you able to compile domoticz yourself and checkout this PR?

https://github.com/domoticz/domoticz/pull/6521

From what I tested this seems to work now..

**Comment by gizmocuz** [comment]
This should now work

**Comment by megamarco833** [comment]
> This should now work

i tested in domoticz beta version build 17277 on raspberry pi in docker.
i see this:

<img width="2042" height="519" alt="Image" src="https://github.com/user-attachments/assets/ed100939-2c53-4c21-8ee5-daa056dbd914" />

and now it's reported the script name that trigger the device, but i'm pretty sure that is reported wrong.
to me, based on the tests that i did, the device was triggered by thermostat dzvent script, not by solar_optimization_5 dzvents script.
can you also please check it?

in my case both scripts runs every 1minutes and both can trigger the same device, but of course in different conditions, and the ones that should trigger that device was the thermostat
